src/rpc/util.cpp

```diff
@@ -634,6 +636,56 @@ UniValue RPCHelpMan::HandleRequest(const JSONRPCRequest& request) const
     return ret;
 }
 
+namespace detail {
+const UniValue* MaybeArg(std::function<void(const RPCArg&)> check, const std::vector<RPCArg>& params, const JSONRPCRequest* req, size_t i)
+{
+    CHECK_NONFATAL(i < params.size());
+    const UniValue& arg{CHECK_NONFATAL(req)->params[i]};
+    const RPCArg& param{params.at(i)};
+    check(param);
+
+    if (!arg.isNull()) return &arg;
+    if (!std::holds_alternative<RPCArg::Default>(param.m_fallback)) return nullptr;
+    return &std::get<RPCArg::Default>(param.m_fallback);
+}
+} // namespace detail
+
+static void CheckNoDefault(const RPCArg& param)
+{
+    // Must use `MaybeArg<Type>(i)` to get the argument.
+    CHECK_NONFATAL(!std::holds_alternative<RPCArg::Default>(param.m_fallback));
```

Comment:
Shouldn't this be `!required && !holds_alternative` ? Why use `MaybeArg(1)` instead of `Arg(1)` for a required arg?

(One reason: maybe we want to be able to change the `RPCHelpMan` definition for a call at runtime, eg based on `-deprecatedrpc` -- in that case if `deprecatedrpc` turns a required arg into an optional, being able to use `MaybeArg` might make the code simpler. But in that case `MaybeArg` should just always work, and not have a check, I think?)